### PR TITLE
feat: route ingress traffic based on chart version

### PR DESCRIPTION
**Pull request by bernielomax**
Support backwards compatibility for webapp routing between v1 and v2+ charts. In v1 charts, the webapp is served from a dedicated webapp service, while in v2+ charts the webapp is served directly from the server service.

- Add chart version parameter to Ingress function
- Add version-specific routing logic
- Add test coverage for version boundaries and routing behavior

### CODE REVIEW

File: internal/k8s/ingress.go [hunk 1]
```diff
@@ -48,37 +49,83 @@ func Ingress(hosts []string) *networkingv1.Ingress {
 }
 
 // ingressRule creates a rule for the host with proper API routing.
-func ingressRule(host string) networkingv1.IngressRule {
-	var pathType = networkingv1.PathType("Prefix")
+func ingressRules(chartVersion string, host string) networkingv1.IngressRule {
+	rules := ingressRulesForV1()
+	if helm.ChartIsV2Plus(chartVersion) {
```
Comment (bernielomax) [inline internal/k8s/ingress.go:54]: For this we probably want to be checking against Airbyte versions? That's not something we currently do, or truly support.

**Review by bgroff — APPROVED** [review]
